source/slang/slang-parser.cpp

```diff
@@ -5579,6 +5597,144 @@ namespace Slang
         return varDeclrStatement;
     }
 
+    static Expr* constructIfLetPredicate(Parser* parser, VarExpr* varExpr)
+    {
+        // create a "var.hasValue" expression
+        MemberExpr* memberExpr = parser->astBuilder->create<MemberExpr>();
+        memberExpr->baseExpression = varExpr;
+        parser->FillPosition(memberExpr);
+        memberExpr->name = getName(parser, "hasValue");
+
+        // create a "==" operator
+        auto opExpr = parser->astBuilder->create<VarExpr>();
+        opExpr->name = getName(parser, "==");
+        opExpr->scope = parser->currentScope;
+        parser->FillPosition(opExpr);
+
+        // create a 'true' literal
+        BoolLiteralExpr* boolLiteralExpr = parser->astBuilder->create<BoolLiteralExpr>();
+        boolLiteralExpr->value = true;
+
+        // create a "var.hasValue == true" expression
+        InfixExpr* infixExpr = parser->astBuilder->create<InfixExpr>();
+        infixExpr->loc = opExpr->loc;
+        infixExpr->functionExpr = opExpr;
+        infixExpr->arguments.add(memberExpr);
+        infixExpr->arguments.add(boolLiteralExpr);
+        return infixExpr;
+    }
+
+    static VarDecl* parseIfLetExpr(Parser* parser)
+    {
+        Decl* parsedDecl = nullptr;
+        auto identifierToken = peekToken(parser);
+        auto name = identifierToken.getName();
+        auto syntaxDecl = tryLookUpSyntaxDecl(parser, name);
```

Comment:
Don't need any of these.

Just:

ReadToken(TokenType::LParent);
ReadToken("let");
ifLetStmt->varName = readToken(TokenType::Identifier);
ReadToken(TokenType::OpAssign);
ifLetStmt->varExpr = parseExpr();
ReadToken(TokenType::RParent);
ifLetStmt->trueStmt = parseStatement();
if (advanceIf("else"))
    ifLetStmt->falseStmt = parseStatement();

Reply:
done.